src/vtkF3DAlembicImporter.cxx

```diff
@@ -0,0 +1,351 @@
+
+#include <vtkActor.h>
+#include <vtkActorCollection.h>
+#include <vtkCamera.h>
+#include <vtkDoubleArray.h>
+#include <vtkFloatArray.h>
+#include <vtkImageData.h>
+#include <vtkImageReader2.h>
+#include <vtkImageReader2Factory.h>
+#include <vtkLight.h>
+#include <vtkMatrix4x4.h>
+#include <vtkObjectFactory.h>
+#include <vtkPointData.h>
+#include <vtkPoints.h>
+#include <vtkPolyData.h>
+#include <vtkPolyDataMapper.h>
+#include <vtkProperty.h>
+#include <vtkQuaternion.h>
+#include <vtkRenderer.h>
+#include <vtkShaderProperty.h>
+#include <vtkSmartPointer.h>
+#include <vtkStringArray.h>
+#include <vtkTexture.h>
+#include <vtkTransform.h>
+#include <vtkUniforms.h>
+#include <vtkUnsignedShortArray.h>
+
+#include <vtksys/SystemTools.hxx>
+
+#include <Alembic/AbcGeom/All.h>
+// #include <Alembic/AbcCoreHDF5/All.h> Not sure if we want to support legacy HDF5 stuff
+#include <Alembic/AbcCoreOgawa/All.h>
+#include <Alembic/Util/All.h>
+#include <Alembic/Abc/All.h>
+#include <Alembic/AbcCoreAbstract/All.h>
+#include <Alembic/AbcCoreFactory/All.h>
+
+vtkStandardNewMacro(vtkF3DAlembicImporter);
+
+
+class vtkF3DAlembicImporterInternalEx
+{
+	typedef std::map<Alembic::AbcGeom::PlainOldDataType,std::string> PODStringMap;
+	typedef std::map<Alembic::AbcGeom::PropertyType,std::string> PropertyTypeStringMap;
+public:
+	vtkF3DAlembicImporterInternalEx(vtkF3DAlembicImporter* parent) {
+		this->Parent = parent;
+	}
+	void CreatePODStringMap(PODStringMap& podStringMap)
+	{
+		podStringMap[Alembic::AbcGeom::kBooleanPOD] = "kBooleanPOD";
+		podStringMap[Alembic::AbcGeom::kUint8POD] = "kUint8POD";
+		podStringMap[Alembic::AbcGeom::kInt8POD] = "kInt8POD";
+		podStringMap[Alembic::AbcGeom::kUint16POD] = "kUint16POD";
+		podStringMap[Alembic::AbcGeom::kInt16POD] = "kInt16POD";
+		podStringMap[Alembic::AbcGeom::kUint32POD] = "kUint32POD";
+		podStringMap[Alembic::AbcGeom::kInt32POD] = "kInt32POD";
+		podStringMap[Alembic::AbcGeom::kUint64POD] = "kUint64POD";
+		podStringMap[Alembic::AbcGeom::kInt64POD] = "kInt64POD";
+		podStringMap[Alembic::AbcGeom::kFloat16POD] = "kFloat16POD";
+		podStringMap[Alembic::AbcGeom::kFloat32POD] = "kFloat32POD";
+		podStringMap[Alembic::AbcGeom::kFloat64POD] = "kFloat64POD";
+		podStringMap[Alembic::AbcGeom::kStringPOD] = "kStringPOD";
+		podStringMap[Alembic::AbcGeom::kWstringPOD] = "kWstringPOD";
+		podStringMap[Alembic::AbcGeom::kNumPlainOldDataTypes] = "kNumPlainOldDataTypes";
+		podStringMap[Alembic::AbcGeom::kUnknownPOD] = "kUnknownPOD";
+	}
+	void CreatePropertyTypeStringMap(PropertyTypeStringMap& propertyTypeStringMap)
+	{
+		propertyTypeStringMap[Alembic::AbcGeom::kCompoundProperty] = "kCompoundProperty";
+		propertyTypeStringMap[Alembic::AbcGeom::kScalarProperty] = "kScalarProperty";
+		propertyTypeStringMap[Alembic::AbcGeom::kArrayProperty] = "kArrayProperty";
+	}
+	void ProcessIPolyMesh(vtkRenderer* renderer, const Alembic::AbcGeom::IPolyMesh& pmesh)
+	{
+		vtkSmartPointer<vtkPoints> points = vtkSmartPointer<vtkPoints>::New();
```

Comment:
prefer `vtkNew` when the pointer is not modified. `vtkNew` and `vtkSmartPointer` are conceptually equivalent to `std::unique_ptr` and `std::shared_ptr` respectively.

Reply:
I have change to using vtkNew<>